runtime/src/accounts_background_service.rs

```diff
@@ -334,10 +329,10 @@ impl SnapshotRequestHandler {
         }
         flush_accounts_cache_time.stop();
 
-        let hash_for_testing = if test_hash_calculation {
+        let accounts_hash_for_testing = previous_accounts_hash.map(|previous_accounts_hash| {
             let check_hash = false;
```

Comment:
OOS nit: we can just use `check_hash: false` in the `CalcAcountsHashConfg` and its' just as clear, but with fewer lines